client/gutenberg/extensions/giphy/settings.js

```diff
@@ -0,0 +1,47 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import { Path, SVG } from '@wordpress/components';
+
+export const settings = {
+	name: 'giphy',
+	icon: (
+		<SVG xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+			<Path fill="none" d="M0 0h24v24H0V0z" />
+			<Path d="M17.343 9.171h-4.114V5.057H7.057v13.886h10.286V9.17zm2.057-.514V21H5V3h10.286v2.057h2.057v2.057H19.4v1.543z" />
+		</SVG>
+	),
+	prefix: 'jetpack',
+	title: __( 'Giphy' ),
+	category: 'jetpack',
+	keywords: [ __( 'giphy' ) ],
+	description: __( 'Add a giphy graphic.' ),
+	attributes: {
+		align: {
+			type: 'string',
+			default: 'center',
+		},
+		caption: {
+			type: 'string',
+		},
+		giphyUrl: {
+			type: 'string',
+			default: '//giphy.com/embed/ZgTR3UQ9XAWDvqy9jv',
+		},
+		searchText: {
+			type: 'string',
+		},
+		topPadding: {
+			type: 'number',
+			default: 56.2,
+		},
+	},
+	supports: {
+		html: false,
+	},
+	apiKey: 'LtJmY9DnDJUA0',
```

Comment:
This seems problematic.

Reply:
The API key in use here is the one that was used in the Atavist block in production for the past ~4 years. I'm 90% sure it's a production key, although it's immaterial since we will need an A8C key anyway (which I'd forgotten, so thank you!). I'll contact Giphy today about getting one. Unlike Map block, the Giphy key is unrestricted and has no costs associated with it. Given this,  what's your hesitation around committing the key as a setting? The key is visible in the XHR requests anyway, and key theft isn't a concern the way it is for a monetized service.

Reply:
👍 No worries if it's save to use and share publicly and you're on top of it 🙂

Reply:
Perfect, thanks for elaborating, Jeff!

Reply:
Giphy has approved the API for production use, so we can call this resolved.